src/Symfony/Component/Ldap/Security/LdapBadge.php

```diff
@@ -24,20 +24,24 @@
  */
 class LdapBadge implements BadgeInterface
 {
-    private $resolved = false;
+    private $resolved = true;
     private $ldapServiceId;
     private $dnString;
     private $searchDn;
     private $searchPassword;
     private $queryString;
 
-    public function __construct(string $ldapServiceId, string $dnString = '{username}', string $searchDn = '', string $searchPassword = '', string $queryString = null)
+    public function __construct(string $ldapServiceId, string $dnString = '{user_identifier}', string $searchDn = '', string $searchPassword = '', string $queryString = null, $resolved = false)
     {
         $this->ldapServiceId = $ldapServiceId;
         $this->dnString = $dnString;
         $this->searchDn = $searchDn;
         $this->searchPassword = $searchPassword;
         $this->queryString = $queryString;
+        $this->resolved = $resolved;
+        if (false === $this->resolved) {
+            trigger_deprecation('symfony/ldap', '5.4', 'Passing false as resolved initial value is deprecated, use true instead.');
```

Comment:
these deprecations make me thing this PR should target 6.4 instead - it's a new feature / an improvement